File: Benchmarking/src/gen431.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  std::vector<s0> f40(const std::vector<s0> &v0) {
    std::vector<s0> v5 = v0;
    std::vector<s0> v2 = v5;
    const s0 v3 = v0[0];
    s0 v8 = v3;
    const std::vector<std::vector<double>> v7 = v8.p0;
    const std::vector<double> v1 = v7[0];
    v2[0] = v3;
    const std::vector<double> v6 = v7[0];
    v5[1] = v8;
    v8.p0 = v7;
    v2[0] = v8;
    v8.p0 = v7;
    v8.p0 = v7;
    std::vector<double> v10 = v1;
    const std::vector<std::vector<double>> v11 = v3.p0;
    const std::vector<std::vector<double>> v14 = v3.p1;
    const std::vector<std::vector<double>> v12 = v8.p0;
    const std::vector<std::vector<std::vector<double>>> v13 { v11, v7, v12, v11, v11, v7 };
    std::vector<double> v20 = v10;
    std::vector<std::vector<double>> v19 = v7;
    v8.p1 = v14;
    const std::vector<double> v27 = v7[0];
    v8.p1 = v14;
    v5[0] = v3;
    std::vector<std::vector<std::vector<double>>> v17 = v13;
    v19 = v12;
    const std::vector<std::vector<double>> v29 = v3.p1;
    v8.p1 = v14;
    const std::vector<std::vector<double>> v21 = v8.p1;
    const std::vector<std::vector<double>> v24 = v3.p0;
    const std::vector<std::vector<double>> v40 = v17[2];
    std::vector<std::vector<double>> v18 = v40;
    v8.p0 = v24;
    v18[0] = v6;
    v8.p0 = v19;
    const s0 v43 = v0[1];
    s0 v28 = v3;
    v28.p1 = v29;
    v18[0] = v20;
    s0 v31 = v43;
    const double v59 = v10[0];
    const std::vector<std::vector<double>> v30 = v8.p1;
    std::vector<std::vector<double>> v50 = v21;
    v31.p0 = v18;
    const std::vector<std::vector<s0>> v39 { v5, v2, v2, v5, v5, v0 };
    const s0 v72 = v0[1];
    v20 = v1;
    v8.p0 = v40;
    std::vector<std::vector<double>> v63 = v30;
    v63[0] = v1;
    const s0 v56 = v5[1];
    const std::vector<std::vector<double>> v70 = v56.p1;
    v10[0] = v59;
    v31.p1 = v14;
    const std::vector<std::vector<double>> v62 = v72.p1;
    const std::vector<std::vector<double>> v58 = v17[1];
    const std::vector<std::vector<double>> v60 = v17[5];
    v8.p1 = v14;
    v31.p0 = v58;
    v18[0] = v27;
    v28.p1 = v70;
    v31.p1 = v63;
    const std::vector<std::vector<double>> v33 = v28.p0;
    const s0 v76(v58, v62);
    const double v68 = v27[0];
    v8.p0 = v60;
    const std::vector<s0> v77 = v39[2];
    const double v69 = v59 * v68;
    v31.p1 = v30;
    const std::vector<std::vector<double>> v122 = v31.p0;
    v20[0] = v69;
    v8.p0 = v122;
    v63 = v50;
    std::vector<s0> v116 = v2;
    v116[1] = v76;
    v8.p0 = v33;
    v116 = v77;
    return v116;
  }
  s0 f37(const s0 &v0) {
    const std::vector<std::vector<double>> v4 = v0.p1;
    s0 v6 = v0;
    std::vector<std::vector<double>> v5 = v4;
    v6.p1 = v4;
    const std::vector<double> v2 = v5[1];
    v6.p1 = v5;
    v6.p1 = v5;
    const std::vector<std::vector<double>> v3 = v6.p1;
    v5[0] = v2;
    s0 v1 = v6;
    v1.p1 = v3;
    v1 = v0;
    const std::vector<std::vector<double>> v38 = v1.p1;
    v1.p1 = v38;
    v1.p1 = v38;
    s0 v46 = v1;
    v6 = v0;
    return v46;
  }
  s0 f36(const s0 &v0, const s0 &v1) {
    const std::vector<std::vector<double>> v2 = v1.p0;
    std::vector<std::vector<double>> v5 = v2;
    const s0 v8 = f37(v0);
    const std::vector<double> v9 = v2[0];
    const std::vector<std::vector<double>> v15 = v8.p1;
    v5 = v2;
    std::vector<std::vector<double>> v12 = v15;
    v12[0] = v9;
    const std::vector<double> v33 = v12[0];
    std::vector<std::vector<double>> v13 = v5;
    v5[0] = v33;
    std::vector<std::vector<double>> v28 = v13;
    const s0 v70(v28, v12);
    v12[1] = v33;
    v13[0] = v33;
    const std::vector<double> v65 = v28[0];
    v12[0] = v65;
    return v70;
  }
  s0 f35(const s0 &v0, const s0 &v1) {
    const s0 v5 = f36(v0, v0);
    const s0 v4 = f37(v1);
    const s0 v6 = f36(v0, v5);
    const s0 v2 = f37(v5);
    s0 v7 = v0;
    const std::vector<std::vector<double>> v3 = v2.p0;
    const s0 v13 = f37(v5);
    const std::vector<std::vector<double>> v12 = v4.p0;
    const std::vector<std::vector<double>> v15 = v6.p0;
    v7.p0 = v12;
    const s0 v24 = f37(v7);
    const s0 v22 = f36(v5, v4);
    const std::vector<double> v63 = v12[0];
    std::vector<std::vector<double>> v36 = v3;
    const std::vector<std::vector<double>> v32 = v7.p1;
    v36[0] = v63;
    const std::vector<std::vector<double>> v17 = v5.p1;
    v7.p1 = v17;
    const std::vector<std::vector<double>> v28 = v13.p1;
    v7.p0 = v15;
    const std::vector<double> v87 = v28[0];
    std::vector<std::vector<double>> v90 = v36;
    const std::vector<double> v43 = v17[1];
    const std::vector<std::vector<double>> v102 = v22.p0;
    v7.p0 = v90;
    s0 v85 = v24;
    const std::vector<std::vector<double>> v134 = v13.p0;
    v85.p0 = v102;
    v85.p1 = v32;
    const s0 v137 = f36(v85, v85);
    v7.p0 = v90;
    v90[0] = v87;
    const std::vector<std::vector<double>> v65 = v4.p1;
    v36 = v134;
    v90 = v15;
    v7.p1 = v65;
    v36[0] = v43;
    v7.p0 = v36;
    return v137;
  }
  std::vector<s0> f29(const std::vector<s0> &v0) {
    std::vector<s0> v6 = v0;
    const std::vector<s0> v3 = f40(v6);
    const s0 v1 = v0[1];
    const s0 v4 = f37(v1);
    const s0 v13 = v3[1];
    const std::vector<s0> v10 = f40(v3);
    const std::vector<s0> v25 = f40(v10);
    const s0 v41 = f35(v13, v4);
    const std::vector<s0> v22 = f40(v25);
    v6[1] = v41;
    return v22;
  }
  __attribute__((noinline))
  double f0(const std::vector<s0> &v0, const double &v1) {
    const std::vector<s0> v8 = f29(v0);
    const s0 v11 = v8[1];
    const double v3 = v1 + v1;
    const double v30 = v3 + v1;
    const std::vector<std::vector<double>> v20 = v11.p0;
    std::vector<std::vector<double>> v35 = v20;
    const std::vector<double> v40 = v35[0];
    const double v27 = v40[0];
    std::vector<double> v54 = v40;
    const double v37 = v54[0];
    const double v72 = v3 * v37;
    v54[0] = v30;
    v54[0] = v37;
    double v148 = v27;
    v54[0] = v148;
    return v72;
  }
  int main() {
    std::vector<s0> v0({ { { { 0.0 } }, { { 1.0 }, { 2.0 } } }, { { { 3.0 } }, { { 4.0 }, { 5.0 } } } });
    double v1(6.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen431.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  func f40(_ v0: [s0]) -> [s0] {
    var v5: [s0] = v0
    var v2: [s0] = v5
    let v3: s0 = v0[0]
    var v8: s0 = v3
    let v7: [[Double]] = v8.p0
    let v1: [Double] = v7[0]
    v2[0] = v3
    let v6: [Double] = v7[0]
    v5[1] = v8
    v8.p0 = v7
    v2[0] = v8
    v8.p0 = v7
    v8.p0 = v7
    var v10: [Double] = v1
    let v11: [[Double]] = v3.p0
    let v14: [[Double]] = v3.p1
    let v12: [[Double]] = v8.p0
    let v13: [[[Double]]] = [v11, v7, v12, v11, v11, v7]
    var v20: [Double] = v10
    var v19: [[Double]] = v7
    v8.p1 = v14
    let v27: [Double] = v7[0]
    v8.p1 = v14
    v5[0] = v3
    var v17: [[[Double]]] = v13
    v19 = v12
    let v29: [[Double]] = v3.p1
    v8.p1 = v14
    let v21: [[Double]] = v8.p1
    let v24: [[Double]] = v3.p0
    let v40: [[Double]] = v17[2]
    var v18: [[Double]] = v40
    v8.p0 = v24
    v18[0] = v6
    v8.p0 = v19
    let v43: s0 = v0[1]
    var v28: s0 = v3
    v28.p1 = v29
    v18[0] = v20
    var v31: s0 = v43
    let v59: Double = v10[0]
    let v30: [[Double]] = v8.p1
    var v50: [[Double]] = v21
    v31.p0 = v18
    let v39: [[s0]] = [v5, v2, v2, v5, v5, v0]
    let v72: s0 = v0[1]
    v20 = v1
    v8.p0 = v40
    var v63: [[Double]] = v30
    v63[0] = v1
    let v56: s0 = v5[1]
    let v70: [[Double]] = v56.p1
    v10[0] = v59
    v31.p1 = v14
    let v62: [[Double]] = v72.p1
    let v58: [[Double]] = v17[1]
    let v60: [[Double]] = v17[5]
    v8.p1 = v14
    v31.p0 = v58
    v18[0] = v27
    v28.p1 = v70
    v31.p1 = v63
    let v33: [[Double]] = v28.p0
    let v76: s0 = s0(p0: v58, p1: v62)
    let v68: Double = v27[0]
    v8.p0 = v60
    let v77: [s0] = v39[2]
    let v69: Double = v59 * v68
    v31.p1 = v30
    let v122: [[Double]] = v31.p0
    v20[0] = v69
    v8.p0 = v122
    v63 = v50
    var v116: [s0] = v2
    v116[1] = v76
    v8.p0 = v33
    v116 = v77
    return v116
  }
  func f37(_ v0: s0) -> s0 {
    let v4: [[Double]] = v0.p1
    var v6: s0 = v0
    var v5: [[Double]] = v4
    v6.p1 = v4
    let v2: [Double] = v5[1]
    v6.p1 = v5
    v6.p1 = v5
    let v3: [[Double]] = v6.p1
    v5[0] = v2
    var v1: s0 = v6
    v1.p1 = v3
    v1 = v0
    let v38: [[Double]] = v1.p1
    v1.p1 = v38
    v1.p1 = v38
    var v46: s0 = v1
    v6 = v0
    return v46
  }
  func f36(_ v0: s0, _ v1: s0) -> s0 {
    let v2: [[Double]] = v1.p0
    var v5: [[Double]] = v2
    let v8: s0 = f37(v0)
    let v9: [Double] = v2[0]
    let v15: [[Double]] = v8.p1
    v5 = v2
    var v12: [[Double]] = v15
    v12[0] = v9
    let v33: [Double] = v12[0]
    var v13: [[Double]] = v5
    v5[0] = v33
    var v28: [[Double]] = v13
    let v70: s0 = s0(p0: v28, p1: v12)
    v12[1] = v33
    v13[0] = v33
    let v65: [Double] = v28[0]
    v12[0] = v65
    return v70
  }
  func f35(_ v0: s0, _ v1: s0) -> s0 {
    let v5: s0 = f36(v0, v0)
    let v4: s0 = f37(v1)
    let v6: s0 = f36(v0, v5)
    let v2: s0 = f37(v5)
    var v7: s0 = v0
    let v3: [[Double]] = v2.p0
    let v13: s0 = f37(v5)
    let v12: [[Double]] = v4.p0
    let v15: [[Double]] = v6.p0
    v7.p0 = v12
    let v24: s0 = f37(v7)
    let v22: s0 = f36(v5, v4)
    let v63: [Double] = v12[0]
    var v36: [[Double]] = v3
    let v32: [[Double]] = v7.p1
    v36[0] = v63
    let v17: [[Double]] = v5.p1
    v7.p1 = v17
    let v28: [[Double]] = v13.p1
    v7.p0 = v15
    let v87: [Double] = v28[0]
    var v90: [[Double]] = v36
    let v43: [Double] = v17[1]
    let v102: [[Double]] = v22.p0
    v7.p0 = v90
    var v85: s0 = v24
    let v134: [[Double]] = v13.p0
    v85.p0 = v102
    v85.p1 = v32
    let v137: s0 = f36(v85, v85)
    v7.p0 = v90
    v90[0] = v87
    let v65: [[Double]] = v4.p1
    v36 = v134
    v90 = v15
    v7.p1 = v65
    v36[0] = v43
    v7.p0 = v36
    return v137
  }
  func f29(_ v0: [s0]) -> [s0] {
    var v6: [s0] = v0
    let v3: [s0] = f40(v6)
    let v1: s0 = v0[1]
    let v4: s0 = f37(v1)
    let v13: s0 = v3[1]
    let v10: [s0] = f40(v3)
    let v25: [s0] = f40(v10)
    let v41: s0 = f35(v13, v4)
    let v22: [s0] = f40(v25)
    v6[1] = v41
    return v22
  }
  @inline(never)
  func f0(_ v0: [s0], _ v1: Double) -> Double {
    let v8: [s0] = f29(v0)
    let v11: s0 = v8[1]
    let v3: Double = v1 + v1
    let v30: Double = v3 + v1
    let v20: [[Double]] = v11.p0
    var v35: [[Double]] = v20
    let v40: [Double] = v35[0]
    let v27: Double = v40[0]
    var v54: [Double] = v40
    let v37: Double = v54[0]
    let v72: Double = v3 * v37
    v54[0] = v30
    v54[0] = v37
    var v148: Double = v27
    v54[0] = v148
    return v72
  }
  func benchmark() {
    let v0: [s0] = [s0(p0: [[0.0]], p1: [[1.0], [2.0]]), s0(p0: [[3.0]], p1: [[4.0], [5.0]])]
    let v1: Double = 6.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 6.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
